brille/design/Proposed_integrated_syntax_Brill_SpinW_Horace.md

```diff
@@ -0,0 +1,267 @@
+There are two implementations of MSlice: [one](https://github.com/mantidproject/mslice) in Python as part of [Mantid](https://www.mantidproject.org)
+and [the original](http://mslice.isis.rl.ac.uk/Main_Page) written in Matlab.
+The original Matlab MSlice is mostly a GUI program with an undocumented CLI (by using the underlying functions of the GUI).
+This program is also currently not being maintained so we will not use it.
+
+In Python, the syntax to extract data from a powder measurement using Mantid MSlice is:
+
+```python
+import mslice.cli as m
+dat2 = m.Load(nxs_file)
+cut2 = m.Slice(d2, f'|Q|,{qmin},{qmax},{qstep}', f'DeltaE,{emin},{emax},{estep}')
+```
+
+Note that unlike in Horace, the original pixel information is not retained in this rebinning operation.
+
+We propose the following syntax for powder data handling in Horace, should it be implemented:
+
+<table>
+<td><img src="https://placehold.it/5x70/000000/000000?text=+"></td>
+<td>
+<pre lang="matlab">
+proj = projaxes('powder');
+cut2 = h.cut_sqw(sqw_file, proj, [qmin, qstep, qmax], [emin, estep, emax]);
+</pre>
+</td>
+<td><img src="https://placehold.it/5x70/000000/000000?text=+"></td>
+</table>
+
+This syntax should retain pixel information just like the equivalent single crystal invocation.
+
+
+## Modelling the data
+
+We assume that [SpinW](https://spinw.org/) would be used to model the magnetic excitations.
+To set up and evaluate a model on single crystal data with current syntax in Matlab is:
+
+```matlab
+tri = sw_model('triAF', 1); 
+spinw_pars = {'hermit', false, 'formfact', true, 'resfun', 'gauss'}; 
+sim1 = disp2sqw_eval(cut1, @tri.horace_sqw, spinw_pars, fwhm);
+```
+
+This same syntax should also be possible (unchanged) for powder data, which has not been implemented.
+
+In Python it would be:
+
+<table>
+<td><img src="https://placehold.it/5x70/000000/000000?text=+"></td>
+<td>
+<pre lang="python">
+tri = h.sw_model('triAF', 1)
+spinw_pars = {'hermit': False, 'formfact': True, 'resfun': 'gauss'}
+sim1 = h.disp2sqw_eval(cut1, tri.horace_sqw, fwhm, **spinw_pars)
+</pre>
+</td>
+<td><img src="https://placehold.it/5x70/000000/000000?text=+"></td>
+</table>
+
+[//]: # (**comment)
+
+In addition to using SpinW directly, it is possible to use Brille to interpolate a SpinW model (in Matlab):
+
+```matlab
+tri = sw_model('triAF', 1); 
+spinw_pars = {'hermit', false, 'formfact', true, 'resfun', 'gauss'}; 
+brl_pars = {'max_volume', 1e-5};
+brl = brillem.Brille(tri, brl_pars{:});
+sim_cut = disp2sqw_eval(exp_cut, @brl.horace_sqw, spinw_pars, fwhm);
+```
+
+And the proposed Python syntax:
+
+<table>
+<td><img src="https://placehold.it/5x150/000000/000000?text=+"></td>
+<td>
+<pre lang="python">
+import brille as b
+tri = h.sw_model('triAF', 1)
+spinw_pars = {'hermit': False, 'formfact': True, 'resfun': 'gauss'}
+brl_pars = {'max_volume':1e-5}
+brl = b.Brille(tri, **brl_pars)
+sim1 = h.disp2sqw_eval(cut1, brl.horace_sqw, fwhm, **spinw_pars)
```

Comment:
There will be scope for additional data handling; and I do envisage `disp2sqw` being able to call Python use functions. We'll have to discuss how much we want to reimplement Horace functionalities in Python... for a while one end-goal of PACE was a Python implementation of Horace but this was changed to just having a Python interface to Matlab code because we figured it would take too much developer effort to port everything from Matlab to Python.

Reply:
I agree that anything like `horace_sqw` should not be a method of the generic (C++/Python) `brille`.
I guess that this is where my lack of imagination in naming becomes a problem.

In the example MATLAB code above `@brl.horace_sqw` is a method of the `brillem.Brille` class, which is a MATLAB class to handle setting-up and interacting with a generic `brille` interpolation grid through the `brille` Python bindings.

The proposed Python interface would need an interface similar to (MATLAB) `brillem.Brille`. This new interface could be in a submodule of the `brille` Python interface but should probably defined in a new Python module (not unlike `brilleu`).